### PR TITLE
WT-10584 Add missing read barriers in __cursor_skip_prev

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Review by agorrod — APPROVED** [review]
I didn't review the comments, but the code changes here look correct to me.

**Review by korteland — APPROVED** [review]
LGTM, thanks for the example in the comments.

**Comment by quchenhao** [comment]
evergreen merge

File: src/btree/bt_curprev.c [hunk 1]
```diff
@@ -98,11 +104,26 @@ __cursor_skip_prev(WT_CURSOR_BTREE *cbt)
...
+         */
+        WT_ORDERED_READ(next_ins, ins->next[i]);
+        if (next_ins != current) /* Stay at this level */
+            ins = next_ins;
         else { /* Drop down a level */
+            /*
+             * It is possible that we read an old value that is inconsistent to the higher levels of
+             * the skip list due to CPU read reordering. Add a read barrier to avoid this issue.
+             */
+            WT_ORDERED_READ(cbt->next_stack[i], ins->next[i]);
```
Comment (dgomezferro) [inline src/btree/bt_curprev.c:125]: Wouldn't it be better to assign the cached `next_ins` here? We already took care of avoiding the memory reordering.

`cbt->next_stack[i] = next_ins;`

**Review by dgomezferro — COMMENTED** [review]
Just a small suggestion.

File: src/btree/bt_curprev.c (hunk 1 above)
Comment (quchenhao) [reply src/btree/bt_curprev.c:125]: Yes, that's a good point. Will create a ticket to change later

**Pull request merged**